pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -2384,10 +2444,27 @@ default 'raise'
         Return the day of the week represented by the date.
 
         Monday == 1 ... Sunday == 7.
+
+        Examples
+        --------
+        >>> ts = pd.Timestamp('2023-01-01 10:00:00')
+        >>> ts
+        Timestamp('2023-01-01 10:00:00')
+        >>> ts.isoweekday()
+        7
         """
         # same as super().isoweekday(), but that breaks because of how
         #  we have overridden year, see note in create_timestamp_from_ts
-        return self.weekday() + 1
+        try:
+            _dt = datetime(self.year, self.month, self.day,
+                           self.hour, self.minute, self.second,
+                           self.microsecond, self.tzinfo, fold=self.fold)
+        except ValueError as err:
+            raise NotImplementedError(
+                "isoweekday not yet supported on Timestamps which "
+                "are outside the range of Python's standard library. "
+            ) from err
+        return _dt.weekday() + 1
```

Comment:
wait sorry, why did this one have to change? isn't `self.weekday() + 1` correct?

Reply:
I thought that if we give it a negative date it should raise. So, in the stable version:
for `ts = pd.Timestamp('-2000')` the output is:
`ts.isoweekday()=7`
`ts.weekday()=6`